frost-core/src/frost.rs

```diff
@@ -0,0 +1,329 @@
+//! An implementation of FROST (Flexible Round-Optimized Schnorr Threshold)
+//! signatures.
+//!
+//! If you are interested in deploying FROST, please do not hesitate to consult the FROST authors.
+//!
+//! This implementation currently only supports key generation using a central
+//! dealer. In the future, we will add support for key generation via a DKG,
+//! as specified in the FROST paper.
+//!
+//! Internally, keygen_with_dealer generates keys using Verifiable Secret
+//! Sharing, where shares are generated using Shamir Secret Sharing.
+
+use std::{collections::HashMap, convert::TryFrom, fmt, fmt::Debug};
+
+use hex::FromHex;
+
+pub mod keys;
+pub mod round1;
+pub mod round2;
+
+use crate::{Ciphersuite, Error, Field, Group, Signature};
+
+/// The binding factor, also known as _rho_ (ρ)
+///
+/// Ensures each signature share is strongly bound to a signing set, specific set
+/// of commitments, and a specific message.
+///
+/// <https://github.com/cfrg/draft-irtf-cfrg-frost/blob/master/draft-irtf-cfrg-frost.md>
+#[derive(Clone, PartialEq)]
+struct Rho<C: Ciphersuite>(<<C::Group as Group>::Field as Field>::Scalar);
```

Comment:
Optional idea: maybe a type alias can help here?

```rust
pub type ScalarOf<C: Ciphersuite> = <<C::Group as Group>::Field as Field>::Scalar;
```

Reply:
I like this, I may also do it for `Element`; do you have any more ideas on naming of type aliases like this in general for Rust, my default instinct is to call it `Scalar<C>` or `Element<C>` but I worry that might be confusing...

Reply:
The fact that using either of them must be generic over `C` (eg `fn thing<C: Ciphersuite>() -> Scalar<C>` is quite clear to me but you know